modules/virt-accessing-vm-yaml-ssh.adoc

```diff
@@ -0,0 +1,167 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vm-yaml-ssh_{context}"]
+= Accessing a virtual machine via SSH with YAML configurations
+
+You can enable a SSH connection to a virtual machine (VM) without the need to run the `virtctl expose` command. When the YAML file for the VM and the YAML file for the service are configured and applied, the service forwards the SSH traffic to the VM.
+
+The following examples show the configurations for the VM's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* Create a namespace for the VM's YAML file by using the `oc create namespace` command and specifying a name for the namespace.
+
+.Procedure
+. In the YAML file for the VM, add the label and a value for exposing the service for SSH connections. Enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  namespace: ssh-ns <1>
+  name: vm-ssh
+spec:
+  running: false
+  template:
+   metadata:
+    labels:
+     kubevirt.io/vm: vm-ssh <2>
+  spec:
+   domain:
+    devices:
+     disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {} <3>
+        name: testmasquerade <4>
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+  volumes:
+  - name: containerdisk
+    containerDisk:
+      image: kubevirt/fedora-cloud-container-disk-demo
+  - name: cloudinitdisk
+    cloudInitNocloud:
+      userData: |
+        #!/bin/bash
+        echo "fedora" | passwd fedora --stdin
+...
```

Comment:
This VM manifest is wrong - the correct label is missing, the lines are wrongly aligned, etc.
Once again - please use this VM yaml:
```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  namespace: ssh-ns
  name: vm-ssh
spec:
  running: false
  template:
    metadata:
      labels:
        kubevirt.io/vm: vm-ssh
        special: vm-ssh
    spec:
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: containerdisk
          - disk:
              bus: virtio
            name: cloudinitdisk
          interfaces:
          - masquerade: {}
            name: testmasquerade
          rng: {}
        machine:
          type: ""
        resources:
          requests:
            memory: 1024M
      networks:
      - name: testmasquerade
        pod: {}
      volumes:
        - name: containerdisk
          containerDisk:
            image: kubevirt/fedora-cloud-container-disk-demo
        - name: cloudinitdisk
          cloudInitNoCloud:
            userData: |
              #!/bin/bash
              echo "fedora" | passwd fedora --stdin
```

Reply:
Reviewed YAML with Yossi and YAML was updated with spacing issues.

Reply:
Reviewed the YAML together with Yossi in our meeting.